### PR TITLE
Add extraction to resource in CloudSecurity

### PR DESCRIPTION
**Pull request by rleighton**
## Background

Resources often have embedded indicators. This PR adds extraction to walk the resource and extract the `p_any` fields.

## Changes

- Add extractor by reviewing the resource models
- Refactor extractor for CloudTrail

## Testing

- mage test:ci
- mage deploy

### CODE REVIEW

File: internal/log_analysis/log_processor/parsers/awslogs/indicators.go [hunk 1]
```diff
@@ -38,22 +38,7 @@ func mustBuildEventSchema(schema interface{}) interface{} {
 }
 
 func ExtractRawMessageIndicators(w pantherlog.ValueWriter, messages ...pantherlog.RawMessage) {
```
Comment (kostaspap) [inline internal/log_analysis/log_processor/parsers/awslogs/indicators.go:40]: You can just delete this function - any callers of this function should now just call `pantherlog.ExtractRawMessageIndicators`

File: internal/compliance/snapshotlogs/resource.go [hunk 2]
```diff
@@ -52,11 +57,60 @@ type Resource struct {
...
+		case "ID":
+			pantherlog.ScanAWSInstanceID(w, value)
+		case "AccountId", "OwnerId":
+			pantherlog.ScanAWSAccountID(w, value)
+		case "Address", "AssignPublicIp", "PrivateIpAddress", "PrivateIPAddress", "PublicIpAddress", "PublicIPAddress":
+			pantherlog.ScanIPAddress(w, value)
+		case "Domain", "DomainName", "DNSName", "FQDN", "PrivateDnsName", "PublicDnsName":
+			if value != "" {
+				w.WriteValues(pantherlog.FieldDomainName, value)
+			}
```
Comment (kostaspap) [inline internal/compliance/snapshotlogs/resource.go:106]: You can use the pantherlog method for that: 
```
case "Domain", "DomainName", "DNSName", "FQDN", "PrivateDnsName", "PublicDnsName": 
pantherlog.ScanHostname(w, value)

```

File: internal/compliance/snapshotlogs/resource.go [hunk 3]
```diff
@@ -52,11 +57,60 @@ type Resource struct {
 	ARN              pantherlog.String      `json:"arn" panther:"aws_arn" description:"The Amazon Resource Name (ARN) of the resource."`
 	Name             pantherlog.String      `json:"name" description:"The AWS resource name of the resource."`
 	Tags             map[string]string      `json:"tags" description:"A standardized format for key/value resource tags."`
+
+	// a dummy fields that are not exported but used to ensure that `p_any` columns are created in the schema for extracted fields
+	DummyDomain pantherlog.String `json:"-" panther:"domain"`
```
Comment (kostaspap) [inline internal/compliance/snapshotlogs/resource.go:62]: So these are needed? This means we have another bug, we are missing the `p_any_aws_tags` column. 
Can you add another field: 
```
DummyAwsTag pantherrlog.String `json:"-" panther:"aws_tag"`
```

**Review by kostaspap — COMMENTED** [review]
Can you add a new test or extend existing parser tests in `snapshotlogs_test.go` to verify the extraction works as intended?

File: internal/compliance/snapshotlogs/resource.go (hunk 3 above)
Comment (rleighton) [reply internal/compliance/snapshotlogs/resource.go:62]: Not a bug, by design but I'll defer to @alxarch for the "right" way to declare `p_any` columns when they are NOT referenced in the parser structs. . For `ConfigJSON` type parsers, right now we need add dummy struct members. We _could_ change that and allow a `Schema` to be passed like we do the `Config` type parsers, see: https://github.com/panther-labs/panther/blob/master/internal/log_analysis/log_processor/logtypes/entry.go#L116

Comment (rleighton) [reply internal/compliance/snapshotlogs/resource.go:62]: I'll add a dummy for the tags.

**Comment by rleighton** [comment]
> Nice work! Can you add a new test or extend existing parser tests in `snapshotlogs_test.go` to verify the extraction works as intended?

Good catch, I intended to but forgot (fighting deployment bugs fixed in the other PR).

File: internal/compliance/snapshotlogs/resource.go (hunk 3 above)
Comment (rleighton) [reply internal/compliance/snapshotlogs/resource.go:62]: Done

File: internal/log_analysis/log_processor/parsers/awslogs/indicators.go (hunk 1 above)
Comment (rleighton) [reply internal/log_analysis/log_processor/parsers/awslogs/indicators.go]: Done

**Comment by rleighton** [comment]
> > Nice work! Can you add a new test or extend existing parser tests in `snapshotlogs_test.go` to verify the extraction works as intended?
> 
> Good catch, I intended to but forgot (fighting deployment bugs fixed in the other PR).

Done

**Review by kostaspap — APPROVED** [review]

File: internal/compliance/snapshotlogs/resource.go (hunk 2 above)
Comment (rleighton) [reply internal/compliance/snapshotlogs/resource.go]: Done

**Pull request merged**